engine/compiler/compiler.go

```diff
@@ -272,16 +273,24 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 		Branch:   args.Build.Target,
 	}
 
+	addStep := func(step *engine.Step) {
+		step.Labels = labels.Combine(stageLabels, labels.FromStep(&drone.Step{
+			Number: len(spec.Steps) + 1,
```

Comment:
`len(spec.Steps) + 1` will not be an incremental value, correct? `Number` would be the same for all steps?